### PR TITLE
github enterprise support for social auth

### PR DESCRIPTION
**Pull request by QingsongYao**
github enterprise support for social auth to use GITHUB_BASE_DOMAIN and GITHUB_API_DOMAIN. The two variables are also used for Github SSO as well so we can assume people will set once during start-up time.

### CODE REVIEW

**Comment by QingsongYao** [comment]
Can someone review this change?

File: src/social_auth/backends/github.py [hunk 1]
```diff
@@ -30,19 +30,27 @@
 from social_auth.utils import dsa_urlopen
 from social_auth.backends import BaseOAuth2, OAuthBackend
 from social_auth.exceptions import AuthFailed
+import  os
+
+GITHUB_BASE_DOMAIN='github.com'
```
Comment (mattrobenolt) [inline src/social_auth/backends/github.py]: We should definitely not have this configured via environment variables. We can move them into normal settings.

**Comment by QingsongYao** [comment]
I fixed the issue to use setting class to get the config settings.

**Comment by dcramer** [comment]
I'm not really sure I see much value in merging this as is. It wont do anything for SaaS customers, and if you enable it, you're choosing to either only use Enterprise (single install), or not, and only use github.com.

**Comment by QingsongYao** [comment]
Yes,  usually people either use github.com or git Enterprise for their onpromise instance.  This will allow people to link sentry issues with GitEnterpise issues works (which does not work today).

**Comment by mandeepbal** [comment]
I'm trying to use Sentry on-premise and can't use it until it has github enterprise support.

**Comment by dcramer** [comment]
I'm closing this as Sentrys mandate is The cloud service and thus patch doesn't help those users.

**Comment by dcramer** [comment]
Of note when our integrations overhaul happens it will be easier to plugin a GHE specific installation.

**Comment by kapilt** [comment]
your saying no to a five line, zero risk patch, because you don't want to support users of opensource sentry?

**Comment by agis** [comment]
> I'm closing this as Sentrys mandate is The cloud service and thus patch doesn't help those users.

@dcramer Forgive me but I didn't understand this sentence. Could you please elaborate?

**Comment by dcramer** [comment]
@agis this solution doesn't help our cloud service and causes other complexity with open source users. We want to solve things in ways that: 1. Are cloud first, 2. And/or Don't create additional maintenance burden (this would be a feature we'd have to immediately deprecate later this year)

**Pull request closed without merging**